Product/Production/Services/CORE_X12DocumentSubmissionCore/src/main/java/gov/hhs/fha/nhinc/corex12/docsubmission/audit/COREX12RealtimeTransforms.java

```diff
@@ -0,0 +1,584 @@
+            AuditDataTransformConstants.EVENT_ID_CODE_SYS_NAME_DOC, CoreX12AuditDataTransformConstants.ACTIVE_PARTICPANT_ROLE_CODE_DESTINATION_DISPLAY_NAME));
+        return participant;
+    }
+
+    protected ParticipantObjectIdentificationType getParticipantObjectIdentificationType(Object msg) {
+        String payloadId = null;
+        byte[] byteArray = null;
+
+        /* Assign ParticipationObjectIdentification */
+        ParticipantObjectIdentificationType participantObject = new ParticipantObjectIdentificationType();
+        // Set the Partipation Object Id (patient id)
+        if (msg != null) {
+            if (msg instanceof COREEnvelopeRealTimeResponse) {
+                payloadId = ((COREEnvelopeRealTimeResponse) msg).getPayloadID();
+            } else if (msg instanceof COREEnvelopeRealTimeRequest) {
+                payloadId = ((COREEnvelopeRealTimeRequest) msg).getPayloadID();
+            } else if (msg instanceof COREEnvelopeBatchSubmission) {
+                payloadId = ((COREEnvelopeBatchSubmission) msg).getPayloadID();
+            } else if (msg instanceof COREEnvelopeBatchSubmissionResponse) {
+                payloadId = ((COREEnvelopeBatchSubmissionResponse) msg).getPayloadID();
+            }
+            participantObject.setParticipantObjectID(payloadId);
+        } else {
+            return participantObject;
+        }
+        byteArray = marshallToByteArray(msg);
+        TypeValuePairType oType = new TypeValuePairType();
+        oType.setValue(byteArray);
+        participantObject.getParticipantObjectDetail().add(oType);
+
+        // Set the Participation Object Typecode
+        participantObject.setParticipantObjectTypeCode(AuditDataTransformConstants.PARTICIPANT_OJB_TYPE_CODE_SYSTEM);
+
+        // Set the Participation Object Typecode Role
+        participantObject.setParticipantObjectTypeCodeRole(CoreX12AuditDataTransformConstants.PARTICIPANT_OJB_TYPE_CODE_ROLE_X12);
+
+        // Set the Participation Object Id Type code
+        CodedValueType partObjIdTypeCode = new CodedValueType();
+        partObjIdTypeCode.setCode(payloadId);
+        partObjIdTypeCode.setCodeSystemName(CoreX12AuditDataTransformConstants.CAQH_x12_CONNECTIVITY_CODED_SYS_NAME);
+        partObjIdTypeCode.setDisplayName(CoreX12AuditDataTransformConstants.CAQH_x12_CONNECTIVITY_CODED_SYS_DISPLAY_NAME);
+        participantObject.setParticipantObjectIDTypeCode(partObjIdTypeCode);
+
+        return participantObject;
+    }
+
+    private static byte[] marshallToByteArray(Object msg) {
+        byte[] bObject = null;
+        COREEnvelopeRealTimeRequest oRequestNoPayload = null;
+        COREEnvelopeRealTimeResponse oResponseNoPayload = null;
+        COREEnvelopeBatchSubmission oBatchRequestNoPayload = null;
+        COREEnvelopeBatchSubmissionResponse oBatchResponseNoPayload = null;
+        javax.xml.namespace.QName xmlqname = null;
```

Comment:
Do we use a QName from a different package somewhere else, or can we import directly?